sw/cheri/tests/sdcard_tests.hh

```diff
@@ -0,0 +1,291 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define CHERIOT_NO_AMBIENT_MALLOC
+#define CHERIOT_NO_NEW_DELETE
+#define CHERIOT_PLATFORM_CUSTOM_UART
+
+#include <assert.h>
+#include <ctype.h>
+#include <stdint.h>
+
+// clang-format off
+#include "../../common/defs.h"
+// clang-format on
+#include <cheri.hh>
+#include <platform-spi.hh>
+
+#include "../common/console.hh"
+#include "../common/filesys-utils.hh"
+#include "../common/platform-pinmux.hh"
+#include "../common/sdcard-utils.hh"
+#include "../common/sonata-devices.hh"
+
+#include "../tests/test_runner.hh"
+
+// Lorem Ipsum sample text.
+#include "lorem_text.hh"
+
+using namespace CHERI;
+
+#define MAX_BLOCKS 0x10
+#define BLOCK_LEN 0x200
+
+// Set this for manual operation rather than automated regression testing.
+static constexpr bool manual = false;
+
+// Set this to true to enable diagnostic logging.
+static constexpr bool logging = false;
+
+// Set this to true to emit the `lorem ipsum` sample text for capture and subsequent
+// writing to a FAT32-formatted microSD card as `LOREM.IPS` within the root directory.
+static constexpr bool emitText = false;
+
+static uint8_t fileBuffer[BLOCK_LEN];
+
+// Dump out a sequence of bytes as hexadecimal and ASCII text.
+static void dump_bytes(const uint8_t *buf, size_t blkBytes, Log &log) {
```

Comment:
Is this the same as the one in the sdcard check? Can we re-use them somehow?

Reply:
There are presently 3 copies, and one has a comment about moving this into console.hh; I'll move it there unless you can think of a better home?

Reply:
I've moved it to console.hh as proposed.

Reply:
Thanks, that sounds good to me.